pkg/apis/apps/v1alpha1/servicebindingrequest_types.go

```diff
@@ -81,6 +81,41 @@ type ApplicationSelector struct {
 	LabelSelector               *metav1.LabelSelector `json:"labelSelector,omitempty"`
 	metav1.GroupVersionResource `json:",inline"`
 	ResourceRef                 string `json:"resourceRef,omitempty"`
+
+	// BindingPath refers to the path in the application workload's schema
+	// where the binding workload would be referenced.
+	// +optional
+	BindingPath *BindingPath `json:"bindingPath,omitempty"`
+}
+
+const (
+	// DefaultContainersPath has the logical path logical path
+	// to find containers on supported objects
+	// Used as []string{"spec", "template", "spec", "containers"}
+	DefaultPathToContainers = "spec.template.spec.containers"
+
+	// DefaultPathToVolumes is the logical path to find volumes on supported objects
+	// used as []string{"spec", "template", "spec", "volumes"}
+	DefaultPathToVolumes = "spec.template.spec.volumes"
+)
+
+// BindingPath defines the path to the field where the binding would be
+// embedded in the workload
+type BindingPath struct {
+	// ContainersPath defines the path to the corev1.Containers reference
+	// Example: "spec.template.spec.containers"
+	// +optional
+	ContainersPath string `json:"containersPath"`
+
+	// VolumesPath defines the path to the corev1.Volumes reference
+	// Example: "spec.template.spec.volumes"
+	// +optional
+	VolumesPath string `json:"volumesPath"`
+
+	// SecretPath defines the path to a string field where
+	// the secret needs to be assigned.
+	// +optional
```

Comment:
Amend the documentation stating if omitted the secret won't be injected.

Reply:
The secret reference path will be added inside the container by default.